giskard/scanner/robustness/base_numerical_detector.py

```diff
@@ -0,0 +1,142 @@
+from typing import Optional, Sequence
+
+import numpy as np
+import pandas as pd
+
+from ...datasets.base import Dataset
+from ...models.base import BaseModel
+from ..issues import Issue, IssueLevel, Robustness
+from ..logger import logger
+
+
+class BaseNumericalPerturbationDetector:
+    """Base class for metamorphic detectors based on numerical feature perturbations."""
+
+    _issue_group = Robustness
+
+    def __init__(
+        self,
+        perturbation_fraction: float = 0.01,
+        threshold: Optional[float] = None,
+        num_samples: Optional[int] = None,
+        output_sensitivity: Optional[float] = None,
+    ):
+        """
+        Parameters
+        ----------
+        perturbation_fraction: float
+            Fractional perturbation to apply to numerical features (default is 1% change).
+        threshold: Optional[float]
+            The threshold for the fail rate, defined as the proportion of samples for which the model
+            prediction has changed. If the fail rate is greater than the threshold, an issue is created.
+        num_samples: Optional[int]
+            The maximum number of samples to use for testing. If not provided, a default number is used.
+        output_sensitivity: Optional[float]
+            For regression models, the maximum relative change in prediction considered acceptable.
+        """
+        self.perturbation_fraction = perturbation_fraction
+        self.threshold = threshold
+        self.num_samples = num_samples
+        self.output_sensitivity = output_sensitivity
+
+    def run(self, model: BaseModel, dataset: Dataset, features: Sequence[str]) -> Sequence[Issue]:
+        """Run the numerical perturbation detector."""
+        numerical_features = [f for f in features if pd.api.types.is_numeric_dtype(dataset.df[f])]
```

Comment:
This check is not enough, because sometimes categorical features are represented with numerical types (e.g. integers). The more reliable way would be to extract this from the `dataset.column_types`.


```suggestion
        numerical_features = [f for f in features if dataset.column_types[feature] == "numeric"]
```